spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala

```diff
@@ -263,7 +263,37 @@ trait SparkExpressionConverter {
               blockStatement(convertStatement(child, fields) + "AS" + cast)
           case _ => convertStatement(child, fields)
         }
-
+      case ShiftLeft(child, position) =>
+        convertStatement(child, fields) + ConstantString("<<") + convertStatement(position, fields)
+      case ShiftRight(child, position) =>
+        convertStatement(child, fields) + ConstantString(">>") + convertStatement(position, fields)
+      case CaseWhen(branches, elseValue) =>
+        ConstantString("CASE") +
+          makeStatement(branches.map(whenClauseTuple =>
+            ConstantString("WHEN") + convertStatement(whenClauseTuple._1, fields) + ConstantString("THEN") + convertStatement(whenClauseTuple._2, fields)
+          ), "") +
+          {
+            elseValue match {
+              case Some(value) =>
+                ConstantString("ELSE") + convertStatement(value, fields)
+              case None =>
+                EmptyBigQuerySQLStatement()
+            }
+          } + ConstantString("END")
+      case ScalarSubquery(plan, _, _) =>
+        blockStatement(new BigQueryStrategy(this, expressionFactory, sparkPlanFactory).generateQueryFromPlan(plan).get.getStatement())
+      case Coalesce(columns) =>
+        ConstantString(expression.prettyName.toUpperCase) + blockStatement(makeStatement(columns.map(convertStatement(_, fields)), ", "))
+      case If(predicate, trueValue, falseValue) =>
+        ConstantString(expression.prettyName.toUpperCase) + blockStatement(convertStatement(predicate, fields) + "," + convertStatement(trueValue, fields) + "," + convertStatement(falseValue, fields))
```

Comment:
Even though this works, we can shorten it to 
```
case If(child, trueValue, falseValue) =>
        ConstantString("IF") +
          blockStatement(
            convertStatements(fields, child, trueValue, falseValue)
          )
```

Reply:
Addressed